docs/architecture/performance-tests.md

```diff
@@ -0,0 +1,131 @@
+
+## Watch Out for Queues
+While comparing loggen results can be sufficient for A/B testing, it is not enough to accurately estimate the syslog ingestion throughput of the entire system.
+
+In the following example, loggen was able to send 4.3 mln messages in one minute; however, Splunk indexers required an additional two minutes to process these messages. During that time, SC4S processed the messages and stored them in a queue while waiting for the HEC endpoint to accept new batches.
+
+| Splunk Indexers | Total Processing Time (4.3 mln messages) | Estimated Max EPS |
+|-----------------|------------------------------------------|-------------------|
+| 3               | 3 min                                    | 22K               |
+| 30              | 1 min (no delay)                         | 72K               |
+
+When running your tests, make sure to monitor the queues. The easiest way to do this is by accessing your server’s SC4S container and running:
+```bash
+watch "syslog-ng-ctl stats | grep '^dst.\+\(processed\|queued\|dropped\|written\)'"
+```
+
+If the destination is undersized or connections are slow, the number of queued events will increase, potentially reaching thousands or millions. Buffering is an effective solution for handling temporary data peaks, but constant input overflows will eventually fill up the buffers, leading to disk or memory issues or dropped messages. Ensure that you assess your SC4S capacity based on the number of messages that can be processed without putting undue pressure on the buffers.
+
+## Check Your TCP Performance
+Run the following command:
+```
+loggen --interval 60 --rate 120000 -s 800 --no-framing --inet --active-connections=10 <SC4S_IP> 514
+```
+Over a span of 60 seconds, loggen will establish 10 concurrent TCP connections to SC4S and attempt to generate up to 120,000 messages per second for each connection, with each message being 800 bytes in size. The more efficient the SC4S instance, the higher the average rate.
+
+Example results:
+
+* Loggen - c5.2xlarge
+* SC4S(3.29.0) + podman - c5.4xlarge
+* default configuration
+* Splunk Cloud 9.2.2403.105 - 30IDX
+
+| Metric       | Default SC4S        | Finetuned SC4S      |
+|--------------|---------------------|---------------------|
+| Average Rate | 72,153.75 msg/sec   | 115,276.92 msg/sec  |
+
+For more information, refer to [Finetune SC4S for TCP](tcp-optimization.md).
+
+## Check Your UDP Performance
+Run the following command:
+```bash
+loggen --interval 60 --rate 22000 -s 800 --no-framing --dgram <SC4S_IP> 514
+```
+
+Over a span of 60 seconds, loggen will attempt to generate 20,000 logs per second, each 800 bytes in size, which will be sent via UDP.
+
+After running the command, count the number of events that reached Splunk. Since UDP is a lossy protocol, messages can be lost anywhere along the path.
```

Comment:
After running the command, count the number of events that reached Splunk. Since UDP is prone to data loss, messages can be lost anywhere along the path.